common-npm-packages/docker-common-v2/registryauthenticationprovider/acrauthenticationtokenprovider.ts

```diff
@@ -42,7 +45,88 @@ export default class ACRAuthenticationTokenProvider extends AuthenticationTokenP
         if(this.registryURL && this.endpointName) {      
             return new RegistryAuthenticationToken(tl.getEndpointAuthorizationParameter(this.endpointName, 'serviceprincipalid', true), tl.getEndpointAuthorizationParameter(this.endpointName, 'serviceprincipalkey', true), this.registryURL, "ServicePrincipal@AzureRM", this.getXMetaSourceClient());
         }
-
         return null;
     }
+
+    public async getToken(): Promise<RegistryAuthenticationToken> {
+        let authType: string;
+        // Will error out with an internal error if the parameter is not found. This error is determined inside of the
+        // tl.getEndpointAuthorizationScheme/tl.getEndpointAuthorizationParameter and cannot be caught here as it is a
+        // custom error.
+        try {
+            authType = tl.getEndpointAuthorizationScheme(this.endpointName, false);
+        } catch (error) {
+        }
+        try {
+            authType = tl.getEndpointAuthorizationParameter(this.endpointName, "scheme", false);
```

Comment:
Do you need to call this only if getEndpointAuthorizationScheme errors out ? Otherwise, this will overwrite the value even if the former call succeeds ?